src/Stratis.SmartContracts.Core.Validation/Validators/Method/NewObjectValidator.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections.Generic;
+using System.Linq;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Stratis.ModuleValidation.Net;
+
+namespace Stratis.SmartContracts.Core.Validation
+{
+    /// <summary>
+    /// Validates that a <see cref="Mono.Cecil.MethodDefinition"/> does not create new objects
+    /// </summary>
+    public class NewObjectValidator : IMethodDefinitionValidator
+    {
+        public static readonly string ErrorType = "Use Of New";
+        
+        private IEnumerable<string> typeWhitelist;
+
+        public NewObjectValidator(IEnumerable<string> typeWhitelist)
+        {
+            this.typeWhitelist = typeWhitelist;
+        }
+
+        public IEnumerable<ValidationResult> Validate(MethodDefinition method)
+        {
+            if (method.Body?.Instructions == null)
+                return Enumerable.Empty<MethodDefinitionValidationResult>();
+
+            var errors = new List<ValidationResult>();
+
+            var newObjs = method.Body.Instructions.Where(i => i.OpCode == OpCodes.Initobj || i.OpCode == OpCodes.Newobj);
+
+            foreach (var newObj in newObjs)
```

Comment:
Can we not perhaps have 2 foreach's here and be more explicit as what we are validating? ie.

`foreach (var newObj in newObjs.OfType<TypeReference>())`

and 

`foreach (var newObj in newObjs.OfType<MethodReference>())`